plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -211,17 +211,80 @@ public void testAggregationPushdown()
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "test_case_sensitive_aggregation_pushdown",
+                "(a_string varchar(1), a_char char(1), a_bigint bigint)",
+                ImmutableList.of(
+                        "'A', 'A', 1",
+                        "'B', 'B', 2",
+                        "'a', 'a', 3",
+                        "'b', 'b', 4"))) {
+            // case-sensitive functions prevent pushdown
+            assertConditionallyPushedDown(
+                    getSession(),
+                    "SELECT max(a_string), min(a_string), max(a_char), min(a_char) FROM " + table.getName(),
+                    expectAggregationPushdown,
+                    aggregationOverTableScan)
+                    .skippingTypesCheck()
+                    .matches("VALUES ('b', 'A', 'b', 'A')");
+            // distinct over case-sensitive column prevents pushdown
+            assertConditionallyPushedDown(
+                    getSession(),
+                    "SELECT distinct a_string FROM " + table.getName(),
+                    expectAggregationPushdown,
+                    groupingAggregationOverTableScan)
+                    .skippingTypesCheck()
+                    .matches("VALUES 'A', 'B', 'a', 'b'");
+            assertConditionallyPushedDown(
+                    getSession(),
+                    "SELECT distinct a_char FROM " + table.getName(),
+                    expectAggregationPushdown,
+                    groupingAggregationOverTableScan)
+                    .skippingTypesCheck()
+                    .matches("VALUES 'A', 'B', 'a', 'b'");
+            // case-sensitive grouping sets prevent pushdown
+            assertConditionallyPushedDown(getSession(),
+                    "SELECT a_string, count(*) FROM " + table.getName() + " GROUP BY a_string",
+                    expectAggregationPushdown,
+                    groupingAggregationOverTableScan)
+                    .skippingTypesCheck()
+                    .matches("VALUES ('A', BIGINT '1'), ('a', BIGINT '1'), ('b', BIGINT '1'), ('B', BIGINT '1')");
+            assertConditionallyPushedDown(getSession(),
+                    "SELECT a_char, count(*) FROM " + table.getName() + " GROUP BY a_char",
+                    expectAggregationPushdown,
+                    groupingAggregationOverTableScan)
+                    .skippingTypesCheck()
+                    .matches("VALUES ('A', BIGINT '1'), ('B', BIGINT '1'), ('a', BIGINT '1'), ('b', BIGINT '1')");
+
+            // case-insensitive functions can still be pushed down as long as grouping sets are not case-sensitive
+            assertThat(query("SELECT count(a_string), count(a_char) FROM " + table.getName())).isFullyPushedDown();
+            assertThat(query("SELECT count(a_string), count(a_char) FROM " + table.getName() + " GROUP BY a_bigint")).isFullyPushedDown();
```

Comment:
let's add a correctness cases 

- `count(DISTINCT a_string)`
- `count(DISTINCT a_string), `count(DISTINCT a_bigint)` (together)

this could help avoid any regressions in #8562 cc @alexjo2144